impeller/renderer/backend/metal/surface_mtl.mm

```diff
@@ -179,6 +179,7 @@
     }
   }
 
+#if ((FML_OS_MACOSX && !FML_OS_IOS) || FML_OS_IOS_SIMULATOR)
```

Comment:
I observed using wonderous that the waitUntilScheduled was only necessary on simulator and (speculatively) on macOS

Reply:
Is that with this patch or without?

If you've observed this without the changes in this patch, can we move this to a separate patch?

Reply:
Yes I can split this out. Though we need both of these changes to see much of an improvement, otherwise the final scheduling just takes longer

Reply:
removed

Reply:
moved here: https://github.com/flutter/engine/pull/42160